compiler_opt/tools/combine_tfa_policies_lib.py

```diff
@@ -0,0 +1,193 @@
+    new_observation = time_step.observation
+    new_observation, switch_tensor = self._process_observation(new_observation)
+    updated_step = ts.TimeStep(
+        step_type=time_step.step_type,
+        reward=time_step.reward,
+        discount=time_step.discount,
+        observation=new_observation)
+
+    def f0():
+      return tf.cast(
+          self.tf_policies[0].distribution(updated_step).action.cdf(0)[0],
+          dtype=tf.float32)
+
+    def f1():
+      return tf.cast(
+          self.tf_policies[1].distribution(updated_step).action.cdf(0)[0],
+          dtype=tf.float32)
+
+    distribution = tf.cond(
+        tf.math.reduce_all(tf.equal(switch_tensor, self.high_low_tensor)), f0,
+        f1)
+    return tf_agents.trajectories.PolicyStep(
+        action=self._create_distribution(distribution), state=policy_state)
+
+
+@gin.configurable()
+def get_input_signature():
+  """Returns a list of inlining features to be used with the combined models."""
+  # int64 features
+  inputs = dict((key, tf.TensorSpec(dtype=tf.int64, shape=(), name=key))
+                for key in [
+                    "caller_basic_block_count",
+                    "caller_conditionally_executed_blocks",
+                    "caller_users",
+                    "callee_basic_block_count",
+                    "callee_conditionally_executed_blocks",
+                    "callee_users",
+                    "nr_ctant_params",
+                    "node_count",
+                    "edge_count",
+                    "callsite_height",
+                    "cost_estimate",
+                    "inlining_default",
+                    "sroa_savings",
+                    "sroa_losses",
+                    "load_elimination",
+                    "call_penalty",
+                    "call_argument_setup",
+                    "load_relative_intrinsic",
+                    "lowered_call_arg_setup",
+                    "indirect_call_penalty",
+                    "jump_table_penalty",
+                    "case_cluster_penalty",
+                    "switch_penalty",
+                    "unsimplified_common_instructions",
+                    "num_loops",
+                    "dead_blocks",
+                    "simplified_instructions",
+                    "constant_args",
+                    "constant_offset_ptr_args",
+                    "callsite_cost",
+                    "cold_cc_penalty",
+                    "last_call_to_static_bonus",
+                    "is_multiple_blocks",
+                    "nested_inlines",
+                    "nested_inline_cost_estimate",
+                    "threshold",
+                    "is_callee_avail_external",
+                    "is_caller_avail_external",
+                ])
+  inputs.update({
+      "model_selector":
+          tf.TensorSpec(shape=(2,), dtype=tf.uint64, name="model_selector")
+  })
+  return ts.time_step_spec(inputs)
+
+
+@gin.configurable()
+def get_action_spec():
```

Comment:
same idea as above - this is something the ProblemConfiguration should be able to provide.